frappe/core/doctype/role_profile/role_profile.py

```diff
@@ -34,24 +34,7 @@ def on_update(self):
 
 	def update_all_users(self):
 		"""Changes in role_profile reflected across all its user"""
-		has_role = frappe.qb.DocType("Has Role")
-		user = frappe.qb.DocType("User")
-
-		all_current_roles = (
-			frappe.qb.from_(user)
-			.join(has_role)
-			.on(user.name == has_role.parent)
-			.where(user.role_profile_name == self.name)
-			.select(user.name, has_role.role)
-		).run()
-
-		user_roles = defaultdict(set)
-		for user, role in all_current_roles:
-			user_roles[user].add(role)
-
-		role_profile_roles = {role.role for role in self.roles}
-		for user, roles in user_roles.items():
-			if roles != role_profile_roles:
-				user = frappe.get_doc("User", user)
-				user.roles = []
-				user.add_roles(*role_profile_roles)
+		users = frappe.get_list("User Role Profile", filters={"role_profile": self.name}, pluck="parent")
```

Comment:
FYI get_list shouldn't be used like this. This is internal code so no perm checks are required. In fact perm checks here is a bug.

Reply:
Noted.